### PR TITLE
Feature/add authorization rules

### PR DESCRIPTION
**Pull request by olanikegloria**
This PR is raised to implement the following tasks:
 Install CanCanCan in your project.
 Add a role column to the users table. Remember to use a migration for this.
 A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
 A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  def initialize(user)
+    # Define abilities for the passed in user here. For example:
+
+    user ||= User.new # guest user (not logged in)
+    if user.role == 'admin'
+      can :manage, :all
+    else
+      can :delete, [Post, Comment], author_id: user.id
+      can :read, :all
+    end
```
Comment (uwadonat) [inline app/models/ability.rb]: - [ ] Let's make sure owner of `Post` and `Comment` can perform all view, update or destroy its own post/comment .
```
else
  can manage, Post, author user
  can manage, Comment, author user
  can :read, :all
  end
end
```

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -4,7 +4,8 @@ def index
       .includes(:comments)
       .where(author: params[:user_id])
       .order(created_at: :asc)
-    @author = @posts.first.author
+    @author = @posts.first.author unless @posts.first.nil?
```
Comment (uwadonat) [inline app/controllers/posts_controller.rb:8]: - [ ] Please uses `CanCanCan` for Post authorization in Post controller.
```
class PostController < ApplicationController
  load_and_authorize_resource
  ...
end
```

**Review by uwadonat — CHANGES_REQUESTED** [review]
Hi @Team, :hand: 

Good job so far! :mechanical_arm: 
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>
### To highlight! :+1: 
- Your commit messages are very descriptive ✔️
- Correct work-flow was used ✔️
- Fair team contribution ✔️
almost

## Status:Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**